src/server/db/migrations/2020-11-13_0.3.0_to_0.4.0/migrate_0.3.0_to_0.4.0.js

```diff
@@ -0,0 +1,414 @@
+};
+
+/**
+ * Configure update, in batch, of the documents
+ *
+ * @param {Object} options - The documents to be updated:
+ * @param {Array} options.updatedDocs - The documents to update
+ *
+ * @return {Promise} - The promise that resolves the update
+ */
+sjrk.storyUpdate.updateDB = function (options) {
+    var details = {
+        dataToPost: options.updatedDocs,
+        responseDataHandler: sjrk.storyUpdate.logUpdateDB
+    };
+
+    return gpii.dbRequest.configureStep(details, options);
+};
+
+/**
+ * Log how many documents were updated
+ *
+ * @param {String} responseString - Response from the database (ignored)
+ * @param {Object} options - Object containing the set of documents:
+ * @param {Array} options.updatedDocs - The documents to update
+ *
+ * @return {Number} - the number of documents updated.
+ */
+sjrk.storyUpdate.logUpdateDB = function (responseString, options) {
+    fluid.log("-- -- --");
+    fluid.log("Updated ", options.updatedDocs.length, " of ", options.totalNumOfDocs, " story documents.");
+    fluid.log("-- -- --");
+    return options.updatedDocs.length;
+};
+
+/**
+ * Create the step that retrieves stories from the database
+ *
+ * @param {Object} options - The documents to be verified:
+ * @param {Array} options.updatedDocs - The documents to verify
+ *
+ * @return {Promise} - The promise that resolves the verificatioon
+ */
+sjrk.storyUpdate.verifyStories = function (options) {
+    var details = {
+        requestUrl: options.storiesUrl,
+        requestErrMsg: "Error retrieving stories from the database: ",
+        responseDataHandler: sjrk.storyUpdate.verifyStoriesData,
+        responseErrMsg: "Error retrieving stories from database: "
+    };
+    return gpii.dbRequest.configureStep(details, options);
+};
+
+/**
+ * Check all database documents for errors related to the data update
+ *
+ * @param {String} responseString - the response from the request to get all documents
+ * @param {Object} options - Where to store any potentially-discovered invalid documents
+ *
+ * @return {Array} - The invalid documents found during verification
+ */
+sjrk.storyUpdate.verifyStoriesData = function (responseString, options) {
+    var allDocs = JSON.parse(responseString);
+
+    if (allDocs.total_rows !== options.totalNumOfDocs) {
+        throw "Number of records retrieved for verification does not match number of records retrieved for updating";
+    }
+
+    var invalidDocs = [];
+
+    if (allDocs.rows) {
+        fluid.each(allDocs.rows, function (aRow) {
+            var aDoc = aRow.doc;
+            // To filter out the "_design/views" doc that doesn't have the "schemaVersion" field
+            if (aDoc.type === "story") {
+
+                // assume all stories were successfully migrated
+                // if an error is discovered, the cause will be tracked
+                var invalidStoryReasons = [];
+
+                if (!aDoc.schemaVersion || aDoc.schemaVersion < sjrk.storyUpdate.newSchemaVersion) {
```

Comment:
If all docs should have the new schema version, shall the 2nd condition be `aDoc.schemaVersion !== sjrk.storyUpdate.newSchemaVersion`?